src/middlewared/middlewared/plugins/device_/device_info_linux.py

```diff
@@ -0,0 +1,105 @@
+            xml = etree.fromstring(output.decode())
+            for child in filter(lambda c: c.get('class') == 'disk', xml.getchildren()):
+                data = {'rotationrate': None}
+                for c in child.getchildren():
+                    if not len(c.getchildren()):
+                        data[c.tag] = c.text
+                    elif c.tag == 'capabilities':
+                        for capability in filter(lambda d: d.text.endswith('rotations per minute'), c.getchildren()):
+                            data['rotationrate'] = capability.get('id')[:-3]
+                lshw_disks[data['logicalname']] = data
+        return lshw_disks
+
+    def get_disk(self, name):
+        disk = self.disk_default.copy()
+        try:
+            block_device = blkid.BlockDevice(os.path.join('/dev', name))
+        except blkid.BlkidException:
+            return disk
+
+        return self.get_disk_details(block_device, disk, self.retrieve_lshw_disks_data())
+
+    @private
+    def get_disk_details(self, block_device, disk, lshw_disks):
+        dev_data = block_device.__getstate__()
+        disk.update({
+            'name': dev_data['name'],
+            'sectorsize': dev_data['io_limits']['logical_sector_size'],
+            'number': ord(dev_data['name'][-1].lower()) - 96,
+            'subsystem': dev_data['name'][:-1],
+        })
+        type_path = os.path.join('/sys/block/', block_device.name, 'queue/rotational')
+        if os.path.exists(type_path):
+            with open(type_path, 'r') as f:
+                disk['type'] = 'SSD' if f.read().strip() == '0' else 'HDD'
+
+        if block_device.path in lshw_disks:
+            disk_data = lshw_disks[block_device.path]
+            if disk['type'] == 'HDD':
+                disk['rotationrate'] = disk_data['rotationrate']
+
+            disk['ident'] = disk['serial'] = disk_data.get('serial', '')
+            disk['size'] = disk['mediasize'] = int(disk_data['size']) if 'size' in disk_data else None
+            disk['descr'] = disk['model'] = disk_data.get('product')
+            # There are some inconsistencies here with linux and freebsd for lunid,
+            # FreeBSD it seems make a DEVICE ID query but I have seen some inconsistency on a WD disk,
+            # please discuss this with mav
+            lun_id_cp = subprocess.Popen(
+                ['sg_vpd', '--quiet', '-i', block_device.path],
+                stdout=subprocess.PIPE, stderr=subprocess.PIPE,
+            )
+            cp_stdout, cp_stderr = lun_id_cp.communicate()
+            if not lun_id_cp.returncode and lun_id_cp.stdout:
+                disk['lunid'] = cp_stdout.strip().split()[0].decode()
+                if disk['lunid'].startswith('0x'):
+                    disk['lunid'] = disk['lunid'][2:]
+
+        if disk['serial'] and disk['lunid']:
+            disk['serial_lunid'] = f'{disk["serial"]}_{disk["lunid"]}'
+
+        return disk
+
+    async def get_valid_zfs_partition_type_uuids(self):
```

Comment:
Why don't we make this shared between OSes? Are there any obstacles to import Linux-created pool in FreeBSD?

Reply:
Good point. Ideally there shouldn't be